### ISSUE TITLE
Incompatible with rand 0.8.x and above

### ISSUE THREAD

**Issue by juhaku** [rpgp/rpgp#143]
I noticed already quite some time ago that when updating `rand` create above `0.8.x` version the implementation will break.

For example below code will not compile due to some traits not being implemented.
```rust
fn encrypt_with_password(compressed: Message, pwd: &str) -> Result<Message, Error> {
    log::debug!("encrypt_with_password, {:?}", compressed);

    let mut rng = thread_rng();
    let s2k = StringToKey::new_default(&mut rng);

    compressed
        .encrypt_with_password(&mut rng, s2k, SymmetricKeyAlgorithm::AES256, || {
            String::from(pwd)
        })
        .map_err(pgp_error_from_pgp)
}
```
This will lead errors such as:
```bash
error[E0277]: the trait bound `ThreadRng: rand_core::CryptoRng` is not satisfied
  --> fmp-common/src/pgp.rs:90:40
   |
90 |     let s2k = StringToKey::new_default(&mut rng);
   |                                        ^^^^^^^^ the trait `rand_core::CryptoRng` is not implemented for `ThreadRng`
   | 
  ::: /home/juha/.cargo/registry/src/github.com-1ecc6299db9ec823/pgp-0.7.2/src/types/s2k.rs:22:27
   |
22 |     pub fn new_default<R: CryptoRng + Rng>(rng: &mut R) -> Self {
   |                           --------- required by this bound in `StringToKey::new_default`

error[E0277]: the trait bound `ThreadRng: rand_core::CryptoRng` is not satisfied
  --> fmp-common/src/pgp.rs:93:32
   |
93 |         .encrypt_with_password(&mut rng, s2k, SymmetricKeyAlgorithm::AES256, || {
   |                                ^^^^^^^^ the trait `rand_core::CryptoRng` is not implemented for `ThreadRng`

error[E0277]: the trait bound `ThreadRng: rand_core::RngCore` is not satisfied
  --> fmp-common/src/pgp.rs:93:32
   |
93 |         .encrypt_with_password(&mut rng, s2k, SymmetricKeyAlgorithm::AES256, || {
   |                                ^^^^^^^^ the trait `rand_core::RngCore` is not implemented for `ThreadRng`
   |
   = note: required because of the requirements on the impl of `rand::Rng` for `ThreadRng`
```
And forcing users to use older version of `rand` crate specifically staying in version `0.7.x`.

Thanks
-Juha

**Comment by dignifiedquire** [comment]
have you tried the latest release? we have recently updated the whole repo to use rand@0.8

**Comment by juhaku** [comment]
Yes I tried, and currently using the latest 0.7.2 version from crates

**Comment by dignifiedquire** [comment]
Sorry, I thought I had already published that update, could you try the current `master` and let me know if this fixes your issue?

**Comment by juhaku** [comment]
Sure, tested it with `master` branch and along with that it needed `zeroize` above `1.4.x` then it works as before. :) latest seemed to be 1.4.2

**Comment by dignifiedquire** [comment]
released `0.8` with support for rand `0.8`
